### DISCUSSION TITLE
BEETWEN on queryRaw

### DISCUSSION THREAD

**Discussion by joseoct** [prisma/prisma#12433] [Q&A]
Considering `layer === 3`
Why does this work
```TS
const result = await this.prisma.$queryRaw`
      SELECT MAX(layer)
      FROM topics
      WHERE layer BETWEEN 3 AND 4
      AND technology_id = ${technology_id}
    `;
```
get 3.9

And this work
```TS
const result = await this.prisma.$queryRaw`
      SELECT MAX(layer)
      FROM topics
      WHERE layer BETWEEN ${layer} AND 4
      AND technology_id = ${technology_id}
    `;
```
get 3.9

But this doesn't work
```TS
const result = await this.prisma.$queryRaw`
      SELECT MAX(layer)
      FROM topics
      WHERE layer BETWEEN ${layer} AND ${layer + 1}
      AND technology_id = ${technology_id}
    `;
```
get null

Schema
```
model Topic {
  id String @id @default(uuid())

  name         String @unique
  explanation  String
  technology_id String
  layer        Float

  created_at DateTime @default(now())
  updated_at DateTime @updatedAt

  technology Technology @relation(fields: [technology_id], references: [id])

  @@map("topics")
}
```

**Comment by austincrim** [comment]
Hey @joseoct 👋 ,

When you say it "doesn't work" are you getting a specific error that you can share with us? Or is your query just not working as expected? What database provider and Prisma version?

**Reply by joseoct** [reply]
No error, I just get null. Considering `layer === 3`
And in the previous cases I get the correct values. Get e.g. 3.9
Postgres, Prisma 3.10.0

**Reply by janpio** [reply]
Can you please open a bug issue so we can reproduce this and confirm if this is a regression @joseoct? Thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
